### PR TITLE
Setup new backend

### PR DESCRIPTION
**Pull request by erikasv**
## 🛠️ Changes
- create endpoints to get data and layer for a given metric (metric was the term defined by Cesar for all "cifras e indicadores" from the search module)
- Read dotenv and disable swagger in production
- Document some of the parameters, the validation for the polygon is more complex, so I'll include it in another issue

## 📝 Associated issues
[187](https://github.com/PEM-Humboldt/biotablero-scrum/issues/187)

## 🤔 Considerations
There is a problem with the length of the line in docstrings, black won't trim them and flake8 will complain... we'll have to discuss other tools or alternatives for this, because the only available option for the moment is to exclude one by one (docstring)

### CODE REVIEW

File: app/routers/metrics.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    return {"layer": "response to be defined"}
+
+
+@router.post("/{metric_id}/layer")
+async def get_layer_by_polygon(
+    metric_id: Annotated[str, fastapi.Depends(metric_id_param)],
+    polygon: Polygon,
+):
+    """
+    Given a metric and a predefined area of interest, get the layer of the metric vut by the indicated area
```
Comment (cazapatamar) [inline app/routers/metrics.py]: Fix typo `vut`

File: app/routers/metrics.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    ]
+
+
+@router.get("/{metric_id}/layer")
+async def get_layer_by_defined_area(
+    metric_id: Annotated[str, fastapi.Depends(metric_id_param)],
+    defined_area: Annotated[dict, fastapi.Depends(defined_areas_params)],
+):
+    """
+    Given a metric and a predefined area of interest, get the layer of the metric vut by the indicated area
```
Comment (cazapatamar) [inline app/routers/metrics.py]: Fix typo `vut`

File: app/main.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+def get_settings():
+    return Settings()
+
+
+settings = get_settings()
+app = FastAPI(
+    title="BioTableroSearch",
+    description="Get metrics by predefined or custom (polygon) areas.",
+    summary="Metrics for BioTablero Search module",
+    version="0.1.0",
```
Comment (cazapatamar) [inline app/main.py:25]: Just to confirm, is the version 0.0.1 or 0.1.0? The README indicates it's version 0.0.1

**Review by cazapatamar — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+python-version = 3.10
```
Comment (ManuelStardust) [inline pyproject.toml]: Excuse me, testing the operation of the actions in stac-data-tools, it's not necessary to put this line with the python version here, it reads it from the configuration for poetry

File: app/routers/metrics.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        description="GeoJSON polygon to determine the query area"
+        # TODO: Add input example
+    )
+
+
+# TODO: Successful response examples for /areas
+# TODO: 422 response examples for /areas and /layer
+
+
+@router.get("/{metric_id}/areas")
```
Comment (cazapatamar) [inline app/routers/metrics.py:44]: Im getting this error

imagen

Comment (cazapatamar) [reply app/routers/metrics.py:44]: http://localhost:8000/metrics/Coverage/areas?area_type=ea&area_id=CAR

**Review by cazapatamar — CHANGES_REQUESTED** [review]

**Review by ManuelStardust — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 3 above)
Comment (erikasv) [reply app/main.py:25]: 0.1.0

File: app/routers/metrics.py (hunk 5 above)
Comment (erikasv) [reply app/routers/metrics.py:44]: fixed

File: pyproject.toml (hunk 4 above)
Comment (erikasv) [reply pyproject.toml]: done

**Review by ManuelStardust — APPROVED** [review]

**Review by cazapatamar — APPROVED** [review]

**Pull request merged**